File: TP 3 - C/verificar_id.c
#include "verificar_id.h"

const int MODO_AGREGAR = 1;
const int MODO_CONTACTAR_MODIFICAR = 2;

bool id_existente(FILE* superheroes, int id, int modo){

    heroe_t heroe;
    bool existe = false;

    int leido = leer_superheroe(superheroes, &heroe);

    while(leido != EOF){
        if(heroe.id == id){
            existe = true;
            if(modo == MODO_AGREGAR){
                printf("La ID ingresada ya existe, pruebe el comando ayuda\n");
            }
        }
        leido = leer_superheroe(superheroes, &heroe);
    }

    if(existe == false && modo == MODO_CONTACTAR_MODIFICAR){
        printf("La ID ingresada no existe, pruebe el comando ayuda\n");
    }

    return existe;
}

File: TP3 - PYTHON/verificar_id.py
from heroe import Heroes
from heroe import leer_superheroe

MODO_AGREGAR = 1
MODO_CONTACTAR_MODIFICAR = 2

#PRE: id > 0 , modo tiene que ser igual a 1 o 2.
#POS: Devuelve true si la id ya existe.
def id_existente(superheroes, id, modo):
    
    heroe = Heroes(0, "", 0, "")
    existe = False

    csvreader = leer_superheroe(superheroes)
    
    for row in csvreader:
        datos = row.strip().split(";")
        heroe.id = datos[0]
        if heroe.id == id:
            existe = True
            if modo == MODO_AGREGAR:
                print("La ID ingresada ya existe, pruebe el comando ayuda")
    
    if(existe == False and modo == MODO_CONTACTAR_MODIFICAR):
        print("La ID ingresada no existe, pruebe el comando ayuda")

    return existe
        

    




